shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -195,10 +187,63 @@ - (NSObject*)valuePublishedByPlugin:(NSString*)pluginKey {
   return nil;
 }
 
-#pragma mark - FlutterAppLifeCycleProvider methods
+#pragma mark - Selectors handling
 
 - (void)addApplicationLifeCycleDelegate:(NSObject<FlutterPlugin>*)delegate {
   [_lifeCycleDelegate addDelegate:delegate];
 }
 
+#pragma mark - UIApplicationDelegate method dynamic implementation
+
+- (BOOL)respondsToSelector:(SEL)selector {
+  if ([_lifeCycleDelegate isSelectorAddedDynamically:selector]) {
+    return [self delegateRespondsSelectorToPlugins:selector];
+  }
+  return [super respondsToSelector:selector];
+}
+
+- (BOOL)delegateRespondsSelectorToPlugins:(SEL)selector {
+  if ([_lifeCycleDelegate hasPluginThatRespondsToSelector:selector]) {
+    return [_lifeCycleDelegate respondsToSelector:selector];
+  } else {
+    return NO;
+  }
+}
+
+- (id)forwardingTargetForSelector:(SEL)aSelector {
+  if ([_lifeCycleDelegate isSelectorAddedDynamically:aSelector]) {
+    [self logCapabilityConfigurationWarningIfNeeded:aSelector];
+    return _lifeCycleDelegate;
+  }
+  return [super forwardingTargetForSelector:aSelector];
+}
+
+// Mimic the logging from Apple when the capability is not set for the selectors.
+// However the difference is that Apple logs these message when the app launches, we only
+// log it when the method is invoked. We can possibly also log it when the app launches, but
+// it will cause an additional scan over all the plugins.
+- (void)logCapabilityConfigurationWarningIfNeeded:(SEL)selector {
+  NSArray* backgroundModesArray =
+      [[NSBundle mainBundle] objectForInfoDictionaryKey:kUIBackgroundMode];
+  NSSet* backgroundModesSet = [[NSSet alloc] initWithArray:backgroundModesArray];
+
+  if (selector == @selector(application:didReceiveRemoteNotification:fetchCompletionHandler:)) {
+    if (![backgroundModesSet containsObject:kRemoteNotificationCapabitiliy]) {
+      NSLog(
+          @"You've implemented -[<UIApplicationDelegate> "
+          @"application:didReceiveRemoteNotification:fetchCompletionHandler:], but you still need "
+          @"to add \"remote-notification\" to the list of your supported UIBackgroundModes in your "
+          @"Info.plist.");
+    }
+  } else if (selector == @selector(application:performFetchWithCompletionHandler:)) {
+    if (![backgroundModesSet containsObject:kBackgroundFetchCapatibility]) {
+      NSLog(@"You've implemented -[<UIApplicationDelegate> "
+            @"application:performFetchWithCompletionHandler:], but you still need to add \"fetch\" "
+            @"to the list of your supported UIBackgroundModes in your Info.plist.");
+    }
+  }
```

Comment:
This feels potentially high-maintenance. If Apple ever changes the SDK to add more of these, we won't know. That said, these aren't grouped into a protocol of any sort that we could check against so I'm not sure there's much we can do about it.

No actual action here, unless you can think of some way to mitigate that.